src/react-components/preferences-screen.js

```diff
@@ -582,6 +588,13 @@ const DEFINITIONS = new Map([
   [
     CATEGORY_MISC,
     [
+      {
+        key: "locale",
+        prefType: PREFERENCE_LIST_ITEM_TYPE.SELECT,
+        options: availableLocales,
+        defaultString: "browser",
+        promptForRefresh: true
```

Comment:
It would be nice not to have to refresh (in part because the refresh text may be in a language the user does not read), but simplest thing is probably best for now. Actually I'm surprised it's necessary, because I thought that enabling this was the goal of `WrappedIntlProvider`. Does that not work the way I thought it would?

Reply:
Don't you just need to specify like `key={locale}` on the `WrappedIntlProvider` component to be sure react rerender the whole tree?

Reply:
The main issue is that `i18n.js` is a really "dumb" chunk of code that is included in a bunch of other files- but it has the duty of loading the locale files via dynamic imports. It somehow needs to be notified that the locale has been changed via preferences in order to load the newly requested locale. This is typically done elsewhere by listening for `statechanged` on the parameter store, however since `i18n.js` is loaded so early, there isn't a reference it can use to the parameter store in order to listen for those events yet. 

Assuming the above is solved, the next step is re-rendering the various `WrappedIntlProvider` trees- but that should be handled automatically as `locale-updated` will be fired, and that will cause the state on `WrappedIntlProvider` to get updated, causing a re-render. I'm not sure that `key={locale}` will do anything more to help with that (maybe that is something from the older versions of `react-intl`?).

Reply:
For the `key={locale}` yeah I read it somewhere, but can't find it in the current documentation. Anyway I saw you're using setState in your component, so this already force a rerender as far as I know.